### PR TITLE
update jna dependency to v4.2.1

### PR DESCRIPTION
**Pull request by marpet**
updating to v 4.2.1 seem to fix the exception:
```
java.lang.UnsatisfiedLinkError: com.sun.jna.Native.pointerSize()I
    at com.sun.jna.Native.pointerSize(Native Method)
    at com.sun.jna.Native.<clinit>(Native.java:88)
    at org.esa.s2tbx.dataio.Utils$CKernel32.<clinit>(Utils.java:87)
    at org.esa.s2tbx.dataio.Utils.GetShortPathName(Utils.java:45)
    at org.esa.s2tbx.dataio.Utils.GetIterativeShortPathName(Utils.java:54)
    at org.esa.s2tbx.dataio.openjpeg.OpenJpegUtils.getTileLayoutWithOpenJPEG(OpenJpegUtils.java:59)
    at org.esa.s2tbx.dataio.s2.Sentinel2ProductReader.retrieveTileLayoutFromGranuleDirectory(Sentinel2ProductReader.java:230)
    at org.esa.s2tbx.dataio.s2.Sentinel2ProductReader.retrieveTileLayoutFromProduct(Sentinel2ProductReader.java:195)
    at org.esa.s2tbx.dataio.s2.Sentinel2ProductReader.updateTileLayout(Sentinel2Pro
```

### CODE REVIEW

**Comment by julienmalik** [comment]
Is it intended that the lib-openjpeg dependency is changed to net.java.dev.jna ?

**Comment by marpet** [comment]
Yes. I think these are the more recent versions. They have changed the groupId.
The current development is on github https://github.com/java-native-access/jna.
There you can see the GAV in the pom

File: pom.xml [hunk 1]
```diff
@@ -270,7 +270,7 @@
             <dependency>
                 <groupId>com.sun.jna</groupId>
```
Comment (julienmalik) [inline pom.xml]: then why it's not changed here ?

**Comment by julienmalik** [comment]
I recently revamped the pom so that only the root pom.xml defines the version of our dependencies. the child pom only declare the artifact name & groupid not the version anymore.

So if I understand well we should change the groupId in both the root pom and the openjpeg pom. + update the version in the root pom.

**Comment by marpet** [comment]
The dependency to jna was declared twice in the root pom. I've changed only one of them. I haven't seen the second one.
What I now noticed is that in the rapideye reader the _net.java.dev.jna goupId_ is already used for a long time. Also in the deimos reader.

**Comment by julienmalik** [comment]
Ok thanks for the notice !

**Comment by marpet** [comment]
Have now removed the duplicated and now unused jna dependency

**Pull request merged**